tests/integration/components/listbox-test.js

```diff
@@ -869,6 +871,33 @@ module('Integration | Component | <Listbox>', function (hooks) {
       // Verify the active option is the previously selected one
       assertActiveListboxOption(getListboxOptions()[0]);
     });
+
+    test('should be possible to open the listbox (with transition) with "Enter"', async function (assert) {
+      await render(hbs`
+        <Listbox as |listbox|>
+           <listbox.Button data-test="headlessui-listbox-button-1">Trigger</listbox.Button>
+           <Transition
+            @show={{listbox.isOpen}}
+           >
+             <listbox.Options
+               @isOpen={{true}}
+               as |options|>
+               <options.Option>
+                Option A
+               </options.Option>
+             </listbox.Options>
+            </Transition>
+         </Listbox>`);
+
+      getListboxButton()?.focus();
+      userEvent.keyboard('{Enter}');
```

Comment:
Unfortunately, `triggerEvent` from ember-helpers does not simulate "Enter" like real browser. Only one handler is executed, thus this test would pass even without a fix for the relevant issue.

Reply:
Yes, similar emulation we used with `dialog` test for [tab press](https://github.com/GavinJoyce/ember-headlessui/blob/master/tests/integration/components/dialog-test.js#L661).